src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -425,19 +428,12 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       }
     }
 
-    def checkNonCyclic(pos: Position, tp: Type, lockedSym: Symbol): Boolean = try {
-      if (!lockedSym.lock(CyclicReferenceError(pos, tp, lockedSym))) false
```

Comment:
`lock` always succeeded. This overload of `checkNonCyclic` is invoked right after `checkNotLocked`.